ddl/callback.go

```diff
@@ -39,14 +46,16 @@ func (bi *BaseInterceptor) OnGetInfoSchema(ctx sessionctx.Context, is infoschema
 type Callback interface {
 	// OnChanged is called after a ddl statement is finished.
 	OnChanged(err error) error
-	// OnSchemaStateChange is called after a schema state is changed.
+	// OnSchemaStateChanged is called after a schema state is changed.
 	OnSchemaStateChanged()
 	// OnJobRunBefore is called before running job.
 	OnJobRunBefore(job *model.Job)
 	// OnJobUpdated is called after the running job is updated.
 	OnJobUpdated(job *model.Job)
 	// OnWatched is called after watching owner is completed.
 	OnWatched(ctx context.Context)
+	// SetDomain is called when the assign the domain to callback.
+	SetDomain(do DomainReloader)
```

Comment:
Could we remove it? I don't see where it would be useful.

Reply:
`ctcHook` and the `defaultHook` need to inherit it to set their own domain because the caller is accessing them by interface function.